src/binary_format.cc

```diff
@@ -37,21 +37,21 @@ void FactorGraph::load_weights(const std::string &filename) {
   std::ifstream file;
   file.open(filename, std::ios::in | std::ios::binary);
 
-  long long count = 0;
-  long long id;
   bool isfixed;
-  char padding;
-  double initial_value;
+  char isfixed_char;
+  weight_id_t count = 0;
+  weight_id_t id;
+  weight_value_t initial_value;
   while (file.good()) {
     // read fields
     file.read((char *)&id, 8);
-    file.read((char *)&padding, 1);
+    file.read((char *)&isfixed_char, 1);
     if (!file.read((char *)&initial_value, 8)) break;
     // convert endian
     id = be64toh(id);
-    isfixed = padding;
-    long long tmp = be64toh(*(uint64_t *)&initial_value);
-    initial_value = *(double *)&tmp;
+    isfixed = isfixed_char;
+    uint64_t tmp = be64toh(*(uint64_t *)&initial_value);
```

Comment:
I think using `weight_value_t` here instead of `uint64_t` is better for keeping byte-width consistency, but it may be misleading due to different byte-order. Btw this `tmp` seems unnecessary, or we should read to `tmp`.